### DISCUSSION TITLE
PIP integration would be great

### DISCUSSION THREAD

**Discussion by ChrisOKay** [PicoQuant/snAPI#3] [Ideas]
It would be great to have the snapi package available via the [package index](https://pypi.org/) (e.g. for version freeze via requirements.txt).
Is this planned in the snapi roadmap?

**Comment by tpoint75** [comment]
There already is a version freeze via requirements.txt! Check out https://github.com/PicoQuant/snAPI/blob/main/dist/snAPI-1.0.1.tar.gz in snAPI-1.0.1\snAPI.egg-info!
The requires.txt only contains numpy and typing. But you also need matplotlib for the demos..

I think we should have it in [package index](https://pypi.org/) for easier updates. However, keep in mind that the package does not run on hardware alone. Therefore, you need to install the device drivers separately. I'm not sure if anyone who installs snAPI directly from the index is wondering why it doesn't work with the hardware.

**Reply by ChrisOKay** [reply]
A valid point. But I would argue that these "some assembly required" issues could be solved by verbose exception messages, pointing towards the required device drivers.

**Comment by tpoint75** [comment]
I try to get the project name:
https://github.com/pypi/support/issues/5899

**Comment by tpoint75** [comment]
pip install --upgrade snapi
chek it out
